fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java

```diff
@@ -121,89 +133,77 @@ public String getLatestVersion(String url) {
             url = url.substring(0, index);
         }
 
-        return resourceType.cast(getResource(findResource(url, version), url, id));
+        return resourceType.cast(getResource(findRegistryResource(resourceType, url, version), url, id));
     }
 
     /**
-     * Determine if the resource associated with the given url exists in the registry.
+     * Get the resources for the given resource type
      *
-     * @param url
-     *     the url of the resource
+     * @param resourceType
+     *     the resource type
      * @return
-     *     true if the resource associated with the given url exists in the registry, false otherwise
+     *     the resources for the given resource type
      */
-    public boolean hasResource(String url) {
-        if (url == null) {
-            return false;
-        }
-
-        String id = null;
-        int index = url.indexOf("#");
-        if (index != -1) {
-            id = url.substring(index + 1);
-            url = url.substring(0, index);
-        }
-
-        String version = null;
-        index = url.indexOf("|");
-        if (index != -1) {
-            version = url.substring(index + 1);
-            url = url.substring(0, index);
-        }
-
-        FHIRRegistryResource resource = findResource(url, version);
-        return (id != null) ? (getResource(resource, url, id) != null) : (resource != null);
+    public <T extends Resource> Collection<T> getResources(Class<T> resourceType) {
+        Objects.requireNonNull(resourceType);
+        return providers.stream()
+                .map(provider -> provider.getRegistryResources(resourceType))
+                .flatMap(Collection::stream)
+                .map(registryResource -> resourceType.cast(registryResource.getResource()))
+                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
     }
 
-
     /**
-     * Unload the resource associated with the given url.
+     * Get the profiles that constrain the given type as a collection of {@link Canonical} URLs
      *
-     * @param url
-     *     the url of the resource
+     * @param type
+     *     the constrained type
```

Comment:
worth explicitly mentioning that this can be a Resource type OR a data type (Or something else?)

Reply:
added to the Javadoc to clarify that I am referring to constrained resource type not general type (resource, complex-type, primitive-type)